### ISSUE TITLE
Update using Series 5 compilers causes crash during triangulate polygon

### ISSUE THREAD

**Issue by millerta** [lanl/LaGriT#58]
Compiled on Ubuntu 16 servers with new ExodusII libs and gcc 5 compilers.
Unsure if this is a build issue or a bug in the code.

/n/swdev/mesh_tools/modulefiles/Ubuntu-16.04-x86_64/lagrit/gcc-5.4.0:
module-whatis	 LAGRIT compiled using gcc 5.4.0 
module		 load gcc/5.4.0 
module		 load exodusii/seacas-2.0/gcc-5.4.0-serial 
setenv		 LAGRIT_SRC_DIR /n/swdev/mesh_tools/lagrit/repos/LaGriT/src 
setenv		 LAGRIT_UTIL_SRC_DIR /n/swdev/mesh_tools/lagrit/repos/LaGriT/lg_util/src 
setenv		 LAGRIT_DIR /n/swdev/mesh_tools/lagrit/install-Ubuntu-16.04-x86_64-gcc5.4.0 
setenv		 LAGRIT_EXE /n/swdev/mesh_tools/lagrit/install-Ubuntu-16.04-x86_64-gcc5.4.0/bin/lagrit 
setenv		 LAGRIT_BUILD_SCRIPT /n/swdev/mesh_tools/lagrit/../config-files/configure-lagrit.sh 
prepend-path	 PATH /n/swdev/mesh_tools/lagrit/install-Ubuntu-16.04-x86_64-gcc5.4.0/bin 

Crash occurs on –O (optimized version) of LaGriT
Runs fine on –g (debug) version of LaGriT
All the test cases run fine.

Crash input/output:

read / polygon.inp / mo_poly_work
cmo / create / mo_pts / / / triplane 
copypts / mo_pts / mo_poly_work 
cmo / select / mo_pts 
triangulate / counterclockwise 
 
polygon.inp:
4 3 0 0 0
1 6.25433769338699 -13.1883812697672 -52.0881456356455
2 -44.5134686266113 -13.1883812697672 -52.0881456356456
3 -41.309768678769 -27.1883812697672 -52.0881456356456
4 6.25433769338698 -27.1883812697672 -52.0881456356455
1 4775 line 1 2
2 4775 line 2 3
3 4775 line 3 4

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x7F56B4D43E08
#1  0x7F56B4D42F90
#2  0x7F56B40F34AF
#3  0x52FEB0 in isleftoron_lg_
#4  0x5301DF in isinteriordiagonal_lg_
#5  0x53045F in maketriangles_lg_
#6  0x530B7C in triangulate_lg_
#7  0x40C170 in msgtty_
#8  0x408226 in control_command_lg_
FIXED in triangulate_lg.f

**Comment by millerta** [comment]
Daniel and I worked on this with Daniel looking into possible compiler and build issues using newer version of gcc and newer Ubuntu 16 servers. Also looked into valgrind output.
Meanwhile I looked for possible code errors.

Fixed bug in triangulate_lg.f
Found by cleaning up pointer and argument declarations.
bug at the location suggested by valgrind and segfault report.
The function isleftoron_lg was missing the last 2 arguments:

Fixed code:

491            if(.not.dup_point) then
492               isresult=isleftoron_lg(x3,y3,x1,y1,x2,y2,isccw,local_eps)
493               point_on_same_side = point_on_same_side.and.isresult
494            endif

Broken code:

         if(.not.dup_point) then
              point_on_same_side = point_on_same_side.and.
     *           isleftoron_lg(x3,y3,x1,y1,x2,y2)
           endif
